u16 register addr no longer builds

Current master fails when `RegisterAddressType` in `create_device!()` and `DeviceInterface::AddressType` in the `basic-register` test are changed from `u8` to `u16`.

The error is as follows:

```Rust
error[E0308]: mismatched types
  --> device-driver/tests/basic-register-16bit-addr.rs:49:1
   |
49 | / device_driver::create_device!(
50 | |     device_name: MyTestDevice,
51 | |     dsl: {
52 | |         config {
...  |
81 | | );
   | | ^
   | | |
   | |_expected `u16`, found `u8`
   |   arguments to this function are incorrect
```

https://github.com/IsaacDynamo/device-driver/blob/94401ea4e77fff3a9fe545b08990b4defb878a5a/device-driver/tests/basic-register-16bit-addr.rs

Using 16bit address works on v1.0.4.

--- a/generation/templates/rust/block.rs.j2
+++ b/generation/templates/rust/block.rs.j2
@@ -180,7 +180,7 @@ impl{{block_generics}} {{ block.name }}{{block_generics}} {
                         {{address_type}},
                         field_sets::{{field_set_name}},
                         ::device_driver::{{access}},
-                    >::new(self.interface(), address as u8, field_sets::{{field_set_name}}::{{reset_value_function}})
+                    >::new(self.interface(), address as {{address_type}}, field_sets::{{field_set_name}}::{{reset_value_function}})
                 {% endwhen %}
                 {% when BlockMethodType::Command { field_set_name_in, field_set_name_out, address_type } %}
                     ::device_driver::CommandOperation::<
@@ -189,7 +189,7 @@ impl{{block_generics}} {{ block.name }}{{block_generics}} {
                         {{address_type}},
                         {{self::get_command_fieldset_name(field_set_name_in)}},
                         {{self::get_command_fieldset_name(field_set_name_out)}},
-                    >::new(self.interface(), address as u8)
+                    >::new(self.interface(), address as {{address_type}})
                 {% endwhen %}
                 {% when BlockMethodType::Buffer { access, address_type } %}
                     ::device_driver::BufferOperation::<

